docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,228 @@ The remaining points can happen any time after 1) is complete although note that
-   * Change the version of Cylc and check the version of Rose used to run the
-     suite.
-   * Prevent suites accidentally upgrading or downgrading versions on restart.
-
-9. Consider `rose prune` functionality.
+8. Consider `rose prune` functionality.
    * Need to make sure it will continue to work correctly.
-   * Need Cylc equivalent but this should be done as part of
+   * Longer term we should build housekeeping into Cylc, see
      [cylc-flow #1159](https://github.com/cylc/cylc-flow/issues/1159).
 
-### CLI
+9. Record version control information when you install a suite, see
+   [cylc-flow #3849](https://github.com/cylc/cylc-flow/issues/3849).
+   * `rose suite-run` current supports this for Subversion only
+     (`svn info` + `svn status` + `svn diff`).
+   * Extend this to support equivalent information for git.
+
+### CLI changes
+
+`cylc install` - new command to install workflows.
+* By default, `cylc install` will install the workflow found in `$PWD` into
+  `~/cylc-run/$(basename $PWD)/runN` (where `runN` = `run1`, `run2` ...).
+* If `run1` already exists it will install into `run2` (and so on).
+  Create a symlink `runN` pointing at the latest run.
+* `--run-name=my-run` implies install into `~/cylc-run/$(basename $PWD)/my-run`.
+  If the target directory already exists then fail.
+* `--no-run-name` implies install into `~/cylc-run/$(basename $PWD)`.
+* `--flow-name=my-flow` implies install into `~/cylc-run/my-flow/runN`.
+* `--directory=/path/to/flow` (`-C ...`) implies install the workflow found in
+  `/path/to/flow` (rather than `$PWD`).
+* All the options accepted by `cylc play` will be accepted by `cylc install`.
+  These options will provide the defaults for when `cylc play` is run with this
+  installed workflow (but they can still be overridden). Relevant options will
+  also apply to other cylc commands when operated on the installed workflow
+  (e.g. `cylc validate`). Note that supporting this will require the creation of
+  a new options file.
+* Installation will involved copying over the files found in the source
+  directory.
+  * This will exclude any `.git`, `.svn`, `log`, share` or `work` directories.
+    Everything else will be copied.
+  * The installation should fail if neither `suite.rc` nor `cylc.flow` exist.
+  * The installation should fail if the target directory is not valid.
+    e.g. you cannot install into `~/cylc-run/my-flow/runN` if `~/cylc-run/my-flow`
+    already contains an installed suite.
+* Version control information will be recorded where relevant (svn/git).
+* The `log, `share`, share/cycle` and `work` directories  will be created
+  following whatever symlink rules are defined for localhost
+  (over-ridable via command line options).
+  * This will continue to support "symlinked alternate run directories"
+    (see [cylc-flow #2935](https://github.com/cylc/cylc-flow/pull/2935)).
+
+`cylc play` - new command to run an installed workflow
```

Comment:
@oliver-sanders and I should get inkscaping when it comes to the docs for this! The cassette player conceit has real potential for helping users understand what is going on and we should exploit that mercilessly.

Reply:
Yes, and ruthlessly too!